changelog/13031.improvement.rst

```diff
@@ -0,0 +1,2 @@
+``pytest.mark.parametrize([], isfunc=...)`` will no longer call the idfunc with the internal ``NOTSET``
```

Comment:
`isfunc`? Is that correct?

Also can you rewrite this from the POV of an user? I mean "NOTSET" is just an internal detail.